src/app/components/shared/menu/link/link.component.ts

```diff
@@ -0,0 +1,111 @@
+import { Location } from "@angular/common";
+import {
+  ChangeDetectionStrategy,
+  Component,
+  Inject,
+  Input,
+  OnChanges,
+} from "@angular/core";
+import { ActivatedRoute, Params } from "@angular/router";
+import { API_ROOT } from "@helpers/app-initializer/app-initializer";
+import {
+  isInternalRoute,
+  MenuLink,
+  MenuRoute,
+} from "@interfaces/menusInterfaces";
+import { Placement } from "@ng-bootstrap/ng-bootstrap";
+
+/**
+ * Menu Link Component
+ */
+@Component({
+  selector: "baw-menu-link",
+  template: `
+    <span
+      [placement]="placement"
+      [ngbTooltip]="tooltip"
+      [class.disabled]="link.disabled"
+    >
+      <ng-container *ngIf="isInternalLink; else external">
+        <!-- Internal Link -->
+        <a
+          class="nav-link"
+          [strongRoute]="internalLink.route"
+          [class.active]="active"
+          [class.disabled]="link.disabled"
+        >
+          <ng-container *ngTemplateOutlet="linkDetails"></ng-container>
+        </a>
+      </ng-container>
+      <ng-template #external>
+        <!-- External Link -->
+        <a class="nav-link" [href]="href" [class.disabled]="link.disabled">
+          <ng-container *ngTemplateOutlet="linkDetails"></ng-container>
+        </a>
+      </ng-template>
+    </span>
+
+    <!-- Link Details -->
+    <ng-template #linkDetails>
+      <div class="icon"><fa-icon [icon]="link.icon"></fa-icon></div>
+      <span id="label">{{ link.label }}</span>
+      <span class="d-none" [id]="id">{{ tooltip }}</span>
```

Comment:
Why does the tooltip need an ID?

Reply:
The ID is to help screen readers, however looking online this doesn't appear to be a standard and I'm not sure where i got the example from. The tooltip generated by ngbTooltip should be sufficient for screen readers i would hope

Reply:
Yeah I don't see how a sequentially repeating auto-id would help.

Oh I see. if ngb is using the `aria-describedby="#some-id"` method of linking the target and the tooltip then this makes sense...

Confirmed:

https://github.com/ng-bootstrap/ng-bootstrap/blob/c332b8ba630fc893aaa3513d68a6a6d14322321d/src/tooltip/tooltip.ts#L205-L218

And looks like it is auto-generated:

https://github.com/ng-bootstrap/ng-bootstrap/blob/c332b8ba630fc893aaa3513d68a6a6d14322321d/src/tooltip/tooltip.ts#L151

And unless I'm misreading this, the `id` input is not even used https://github.com/ng-bootstrap/ng-bootstrap/commit/088363551b5d5ecd602a82aedfc97d43a75a080f